### ISSUE TITLE
Magento installation via metapackage: checkExtensions fails

### ISSUE THREAD

**Issue by oleksii-lisovyi** [magento/magento2#21136]
### Preconditions (*)

1. Magento 2.2.*.

### Steps to reproduce (*)

1. Install Magento via composer as a metapackage (according to [docs](https://devdocs.magento.com/guides/v2.2/install-gde/composer.html)):
```shell
composer create-project --repository=https://repo.magento.com/ magento/project-community-edition <install-directory-name>
```
2. Install package, that has a PHP extension dependency, that is not added to your PHP:
I've installed [aws/aws-sdk-php](https://github.com/aws/aws-sdk-php), for example, that has dependency on `sockets` extension as a dev requirement:
```shell
composer require aws/aws-sdk-php
```
3. Perform Magento installation from root directory:
```shell
php bin/magento setup:install <add all the required options here>
```

### Expected result (*)

Installation finishes successfully.

### Actual result (*)

Installation failed with error:
```
Starting Magento installation:
File permissions check...
[Progress: 1 / 773]
Required extensions check...

In Installer.php line 519:
                                           
  [Exception]                              
  Missing following extensions: 'sockets'  
                                           

Exception trace:
 () at /<magento-root-directory>/setup/src/Magento/Setup/Model/Installer.php:519

```

### Technical details (*)
There is implemented check on the required extensions in the method
`\Magento\Setup\Model\Installer::checkExtensions`, that calls `\Magento\Setup\Model\PhpReadinessCheck::checkPhpExtensions`, that calls `\Magento\Framework\Composer\ComposerInformation::getRequiredExtensions`, that returns list of dependencies from composer.lock file:
```php
        $requiredExtensions = [];
        $allPlatformReqs = array_keys($this->getLocker()->getPlatformRequirements(true));

        if (!$this->isMagentoRoot()) {
            /** @var CompletePackageInterface $package */
            foreach ($this->getLocker()->getLockedRepository()->getPackages() as $package) {
                $requires = array_keys($package->getRequires());
                $requires = array_merge($requires, array_keys($package->getDevRequires()));
                $allPlatformReqs = array_merge($allPlatformReqs, $requires);
            }
        }
        foreach ($allPlatformReqs as $reqIndex) {
            if (substr($reqIndex, 0, 4) === 'ext-') {
                $requiredExtensions[] = substr($reqIndex, 4);
            }
        }
        return array_unique($requiredExtensions);
```
**The root issue:** method `$this->isMagentoRoot()` returns `FALSE` for current Magento installation, because its regular expression `/magento\/magento2...?/` doesn't match `magento/project-community-edition`, that comes from metapackage. 
Most likely it works OK for Magento, because current Magento package name is `magento/magento2ce`, that matches the regular expression above.

**Comment by magento-engcom-team** [comment]
Hi @oleksii-lisovyi. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [ ] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@oleksii-lisovyi do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98219`, `MAGETWO-98220` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @oleksii-lisovyi. Thank you for your report. 
The issue has been fixed in magento/magento2#22116 by @oleksii-lisovyi in 2.3-develop branch
Related commit(s):
 - [2c91e5a431fc3fe6aab36b29205ec2e487993f96](https://github.com/magento/magento2/commit/2c91e5a431fc3fe6aab36b29205ec2e487993f96)

The fix will be available with the upcoming 2.3.3 release.
